packages/ui/components/head-seo/HeadSeo.tsx

```diff
@@ -73,16 +73,17 @@ export const HeadSeo = (props: HeadSeoProps): JSX.Element => {
   // The below code sets the defaultUrl for our canonical tags
 
   // Get the current URL from the window object
-  const url = getBrowserInfo()?.url;
+  const { url, origin } = getBrowserInfo();
   // Check if the URL is from cal.com
   const isCalcom =
     url && (new URL(url).hostname.endsWith("cal.com") || new URL(url).hostname.endsWith("cal.dev"));
   // Get the router's path
   const path = useRouter().asPath;
-  // Build the canonical URL using the router's path, without query parameters. Note: on homepage it omits the trailing slash
-  const calcomCanonical = `https://cal.com${path === "/" ? "" : path}`.split("?")[0];
+  const selfHostedOrigin = origin || "https://cal.com";
   // Set the default URL to either the current URL (if self-hosted) or https://cal.com canonical URL
-  const defaultUrl = isCalcom ? calcomCanonical : url;
+  const defaultUrl = isCalcom
+    ? buildCanonical({ path, origin: "https://cal.com" })
+    : buildCanonical({ path, origin: selfHostedOrigin });
 
```

Comment:
AFAIK this must be generated by the server, to have an effect on the seo, so, I think we should not get the origin from the browser, because this info should be available on build time, SSR or ISR. 

CC. @PeerRich @zomars @keithwillcode @emrysal

Reply:
I agree with this. This is also why the buildCanonical doesn't use this. 

The 'isCalcom' should read the URL ideally on something available on the server.

Reply:
I think we can use `WEBAPP_URL` constant 

`import { WEBAPP_URL } from "@calcom/lib/constants";`

Reply:
@p6l-richard how is this looking?

Reply:
I use the `WEBSITE_URL` instead of the `origin` from getBrowserInfo() to avoid an empty canonical value on SSR